packages/shared/src/hooks/usePostById.ts

```diff
@@ -115,6 +126,52 @@ const usePostById = ({ id, options = {} }: UsePostByIdProps): UsePostById => {
   );
   const post = postById || (options?.initialData as PostData);
 
+  useMutationSubscription({
+    matcher: contentPreferenceMutationMatcher,
+    callback: ({ mutation, variables: mutationVariables, queryClient }) => {
```

Comment:
This is handling optimistic updates of contentPreference state on post author/scout. Ideally we can use `useMutationSubscription` in other places like feed and user listings as well to update cache in those queries.

Reply:
Why do we expose preference in sub elements?
Can't we just rely on the main query and use that as a source of truth rather? so only 1 query needs to validate (you are following User[])?

Reply:
It does not really work like that, most of the entities will get contentPreference from a field, which means it will live in different queries.

Reply:
For example in post modal `contentPreference` will come from feed query and each post item inside the feed. I will add that example next.

Reply:
Hmm I see, ideally it was a sole query, but yeah get that it's not possible :)